# Summer 2022 Data Science Intern Challenge - Question 1

In [35]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings("ignore")

In [2]:
#read the data

df = pd.read_csv("/Users/sandeepreddy/Desktop/Shopify_DS_Challenge/2019 Winter Data Science Intern Challenge Data Set - Sheet1 (1).csv")
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [3]:
# make a copy of the dataset

df_1 = df.copy()
df_1

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0            1       53      746           224            2           cash   
1            2       92      925            90            1           cash   
2            3       44      861           144            1           cash   
3            4       18      935           156            1    credit_card   
4            5       18      883           156            1    credit_card   
...        ...      ...      ...           ...          ...            ...   
4995      4996       73      993           330            2          debit   
4996      4997       48      789           234            2           cash   
4997      4998       56      867           351            3           cash   
4998      4999       60      825           354            2    credit_card   
4999      5000       44      734           288            2          debit   

               created_at  
0     2017-03-13 12:36:56  
1     2017-03-03 17:38:52  
2      2017-03-14 4:23:56  
3     2017-03-26 12:43:37  
4      2017-03-01 4:35:11  
...                   ...  
4995  2017-03-30 13:47:17  
4996  2017-03-16 20:36:16  
4997   2017-03-19 5:42:42  
4998  2017-03-16 14:51:18  
4999  2017-03-18 15:48:18  

[5000 rows x 7 columns]

In [4]:
df_1.order_amount.mean()

3145.128

**Seems like this is how the AOV was calculated.**

In [18]:
# Checking for null values

df_1.isnull().sum()

order_id          0
shop_id           0
user_id           0
order_amount      0
total_items       0
payment_method    0
created_at        0
dtype: int64

In [20]:
# Calculating the Single item value 

df_1['item_value'] = df_1['order_amount']/df_1['total_items']
df_1

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0            1       53      746           224            2           cash   
1            2       92      925            90            1           cash   
2            3       44      861           144            1           cash   
3            4       18      935           156            1    credit_card   
4            5       18      883           156            1    credit_card   
...        ...      ...      ...           ...          ...            ...   
4995      4996       73      993           330            2          debit   
4996      4997       48      789           234            2           cash   
4997      4998       56      867           351            3           cash   
4998      4999       60      825           354            2    credit_card   
4999      5000       44      734           288            2          debit   

               created_at  item_value  
0     2017-03-13 12:36:56       112.0  
1     2017-03-03 17:38:52        90.0  
2      2017-03-14 4:23:56       144.0  
3     2017-03-26 12:43:37       156.0  
4      2017-03-01 4:35:11       156.0  
...                   ...         ...  
4995  2017-03-30 13:47:17       165.0  
4996  2017-03-16 20:36:16       117.0  
4997   2017-03-19 5:42:42       117.0  
4998  2017-03-16 14:51:18       177.0  
4999  2017-03-18 15:48:18       144.0  

[5000 rows x 8 columns]

In [23]:
df_2 = df_1[['shop_id','item_value']]
df_2

shop_id  item_value
0          53       112.0
1          92        90.0
2          44       144.0
3          18       156.0
4          18       156.0
...       ...         ...
4995       73       165.0
4996       48       117.0
4997       56       117.0
4998       60       177.0
4999       44       144.0

[5000 rows x 2 columns]

In [25]:
df_2.item_value.describe()

count     5000.000000
mean       387.742800
std       2441.963725
min         90.000000
25%        133.000000
50%        153.000000
75%        169.000000
max      25725.000000
Name: item_value, dtype: float64

**Seems like a much better understanding compared to the previous value. But this is still not acceptable as the average price of a Sneaker is approximately $388.**

**It can be clearly seen that the maximum value is $25,725. This is affecting our AOV.**
**There could be two possibilities:**

**1. Either the amount was entered in cents rather than in dollars**

**2. Or considering that this is the Richard Mille sneaker (possible for the price mentioned)**

In [51]:
# FIRST Possibility
# Changing the Value from Cents to Dollars (Value/100)

df_2.loc[df_2.item_value == 25725, 'item_value'] = 257.25

In [45]:
print(f"Average Value of Sneakers: {df_2.item_value.mean()}")
print(f"Median Value of Sneakers: {df_2.item_value.median()}")
print(f"Mode Value of Sneakers: {df_2.item_value.mode()}")

Average Value of Sneakers: 153.4395
Median Value of Sneakers: 153.0
Mode Value of Sneakers: 0    153.0
dtype: float64


### In this approach, the Average value is $153

**1b) What metric would you report for this dataset?**

**Solution: "Mean"**

**1c) What is it's value?**

**Solution: $153.4**

In [47]:
# SECOND Possibility

df_3 = df_1[['shop_id','item_value']]
df_3

shop_id  item_value
0          53       112.0
1          92        90.0
2          44       144.0
3          18       156.0
4          18       156.0
...       ...         ...
4995       73       165.0
4996       48       117.0
4997       56       117.0
4998       60       177.0
4999       44       144.0

[5000 rows x 2 columns]

In [49]:
df_3.item_value.describe()

count     5000.000000
mean       387.742800
std       2441.963725
min         90.000000
25%        133.000000
50%        153.000000
75%        169.000000
max      25725.000000
Name: item_value, dtype: float64

In [50]:
print(f"Median Value of Sneakers: {df_3.item_value.median()}")
print(f"Mode Value of Sneakers: {df_3.item_value.mode()}")

Median Value of Sneakers: 153.0
Mode Value of Sneakers: 0    153.0
dtype: float64


### In this approach, since the outlier value is affecting the mean, considering the other central tendency measures, the Average value is $153

### https://www.shopify.com.ng/blog/average-order-value#averageorder

**According to this link, "No measure of Central tendency is the best, but using only one is certainly the Worst" - Taylor**

**1b) What metric would you report for this dataset?**

**Solution: "Median" is better than the mean because it isn't influenced by extremely large values**

**1c) What is it's value?**

**Solution: $153**